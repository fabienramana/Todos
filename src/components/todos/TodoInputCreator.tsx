export default function TodoInputCreator(): JSX.Element{
    return (
        <input className="new-todo" placeholder="What needs to be done?" />
    )
}