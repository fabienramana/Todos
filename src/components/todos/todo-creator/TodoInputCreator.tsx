import React, {useState} from 'react'

type TodoInputCreatorProps = {
    addTodo: (content:string) => void
}

export default function TodoInputCreator({addTodo}: TodoInputCreatorProps){
    const [todoContent, setTodoContent] = useState<string>('');
    


    function handleChange(e: React.ChangeEvent<HTMLInputElement>){
        setTodoContent(e.target.value)
    }

    function handleKeyPress(e: React.KeyboardEvent<HTMLInputElement>){
        if(e.key === 'Enter' && todoContent.trim() !== ""){
            addTodo(todoContent.trim());
            setTodoContent("")
          }
    }


    return (
        <input className="new-todo"
         type="text"
         value={todoContent}
         placeholder="What needs to be done?"
         onChange={handleChange} 
         onKeyDown={handleKeyPress}/>
    )
}