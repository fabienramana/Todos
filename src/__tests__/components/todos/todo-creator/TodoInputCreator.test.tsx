import { render, screen } from '@testing-library/react';
import TodoInputCreator from '../../../../components/todos/todo-creator/TodoInputCreator';
import userEvent from '@testing-library/user-event';


describe('TodoInputCreator component', () => {
  test("should call 'addTodo' with title entered in input if enter key is pressed", () => {

    const addTodo = jest.fn();
    const title = "Coucou"

    render(<TodoInputCreator addTodo={addTodo} />)

    const input = screen.getByPlaceholderText("What needs to be done?")

    userEvent.type(input, `${title}{Enter}`);

    expect(addTodo).toHaveBeenCalledTimes(1);
    expect(addTodo).toHaveBeenCalledWith(title);
  })

  test("shouldn't call 'addTodo' if input is empty", () => {

    const addTodo = jest.fn();

    render(<TodoInputCreator addTodo={addTodo} />)

    const input = screen.getByPlaceholderText("What needs to be done?")

    userEvent.type(input, `{Enter}`);

    expect(addTodo).toHaveBeenCalledTimes(0);
  })

  test("shouldn't call 'addTodo' if input is only whitespaces", () => {

    const addTodo = jest.fn();

    render(<TodoInputCreator addTodo={addTodo} />)

    const input = screen.getByPlaceholderText("What needs to be done?")

    userEvent.type(input, `    {Enter}`);

    expect(addTodo).toHaveBeenCalledTimes(0);
  })

  test('should match snapshot', ()=>{
  
    const addTodo = () => {};

    const {asFragment} = render(<TodoInputCreator addTodo={addTodo} />)

    expect(asFragment()).toMatchSnapshot();
  })
}) 