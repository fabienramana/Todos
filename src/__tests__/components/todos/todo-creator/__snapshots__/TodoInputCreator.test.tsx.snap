// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInputCreator component should match snapshot 1`] = `
<DocumentFragment>
  <input
    class="new-todo"
    placeholder="What needs to be done?"
    type="text"
    value=""
  />
</DocumentFragment>
`;
